### ISSUE TITLE
heading-has-content incorrectly reported for heading tags with static content in React ES6 class components

### ISSUE THREAD

**Issue by ryanswanson** [evcohen/eslint-plugin-jsx-a11y#440]
We have seen some of our heading tags (e.g. `<h4>`) reporting the jsx-a11y/heading-has-content error through ESLint; while others seem fine.  In both cases, the headings have valid static text content.

From our initial diagnosis it seems that it is only mis-reporting for our ES6 class-based React components, while the same use in Stateless Functional Components returns valid.

Here is the full ESLint error:

> Headings must have content and the content must be accessible by a screen reader. (jsx-a11y/heading-has-content)

I've reduced this down to two simple examples - the class version reports the error while the function does not.

Stateless Functional Component
```jsx
import React from 'react';

export default function TestFunction() {
  return <h4>Hi</h4>;
}
```

ES6 Class
```jsx
import React, {Component} from 'react';

export default class TestClass extends Component {
  render() {
    return <h4>Hi</h4>;
  }
}
```

Based on version [6.0.2](https://github.com/evcohen/eslint-plugin-jsx-a11y/tree/v6.0.2).

**Comment by nwalters512** [comment]
I believe this was fixed in https://github.com/evcohen/eslint-plugin-jsx-a11y/pull/437, but a version containing that fix hasn't been released yet.

**Comment by openjck** [comment]
I'm also seeing this with the *anchor-has-content* rule.

**Comment by qw-in** [comment]
#437 should have fixed this for `anchor-has-content` as well as `heading-has-content`. That fix was introduced in `v6.1.0`

@openjck does it still happen in `v6.1.0+`?

**Comment by ljharb** [comment]
Closing; will reopen if needed.

**Comment by openjck** [comment]
@qw-in this is fixed. Here's my commit upgrading the package and removing the lines that disabled these rules.

https://github.com/mozilla/ensemble/commit/5e01cde2eda1b0dd0aa70aae8bf9611a2b1837ec

**Comment by DavideDaniel** [comment]
This is still happening in v6.1.2 for us when header is used with static content.

**Comment by ljharb** [comment]
@DavideDaniel the [commit](https://github.com/taddei/eslint-plugin-jsx-a11y/commit/595b68fb43c3e653b94d95d9f3e8717788d1f70b) that fixes it is still unreleased.

**Comment by ljharb** [comment]
@DavideDaniel can you file a new issue with repro code?
